dll/win32/shlwapi/path.c

```diff
@@ -1151,6 +1151,7 @@ BOOL WINAPI PathFileExistsDefExtW(LPWSTR lpszPath,DWORD dwWhich)
 #ifdef __REACTOS__
         if (dwWhich & 0x1)
         {
+        if (GetFileAttributes(lpszPath) != FILE_ATTRIBUTE_DIRECTORY)
```

Comment:
Just to confirm: By this new if-case, your intent is to just have the lstrcpyW call done below called only when the file is not a directory? (while the PathFileExistsW is always being done?)

Reply:
Yes ,correct. 
Prevent adding  executable extension to directory name